### PR TITLE
[global styling] Introduce global color constants and refactor existing screens/components

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :cloud: :cloud: Description :cloud: :cloud:
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

This PR introduces global color constants and applies them to existing components and screens. This is the first step in efforts to standardize and clean up styling across the repo. With this change, the only use of in-line color constants are in icon SVGs and in the Home screen category card containers (which will be addressed in future cleanup once the designs are finalized).

Naming convention:
- `text_` for any text
- `surface_` for all backgrounds or other surfaces (e.g. text input background color)
- `shadow_` for shadows
- `{color name}_` (e.g. `redPrimary`) for brand colors in the app (red and blue)

On naming conventions, I decided on this scheme (function of the color before the color descriptor e.g.`textPrimary`, `textSecondary`, etc) so that developers can rely on autocomplete to find the colors they need based on the use case. For example, if styling a background, they can just type `surface` and use autocomplete to select the correct color.

image

Along the way, I made some minor tweaks to the colors to make them match a standardized set (after I discussed with Esther on Saturday), so there are some visual changes:
- Most background containers are now white instead of offwhite (as decided by Esther)
- Set the default RectButton to be white text on a red background.

### Screenshots :calling:
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @leexesther at the bottom."
Minor (barely noticeable) changes across the board.

## :zap: How to review :zap:
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Give feedback on color style naming and usage
- Make sure everything still looks right

## :seedling: Next steps :seedling:
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Continue styling standardization with text styles
- Extract hex codes from the home screen

## Relevant Links :link:
n/a

### Online sources :information_source:
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
n/a

### Related PRs :raised_hands:
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
n/a

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @davidqing6432

### CODE REVIEW

File: src/components/RectButton.tsx [hunk 1]
```diff
@@ -7,17 +7,17 @@ import {
...
     textTransform: 'uppercase',
     alignSelf: 'center',
-    color: '#525454',
+    color: Colors.textWhite,
   },
   buttonContainer: {
-    backgroundColor: '#C4C4C4',
+    backgroundColor: Colors.redPrimary,
     width: '40%',
-    height: '5%',
```
Comment (david-qing) [inline src/components/RectButton.tsx:20]: noticed that height being removed makes the buttons hug the text height, kinda hard to press

**Review by david-qing — APPROVED** [review]
